### PR TITLE
Introduce FSDP helper APIs

### PR DESCRIPTION
**Pull request by cbalioglu**
This PR introduces a few small FSDP helper functions, makes some nit improvements in docstrs, and updates the required isort version.

### CODE REVIEW

File: src/fairseq2/nn/utils/module.py [hunk 1]
```diff
@@ -97,20 +130,20 @@ def empty_like(src: Tensor) -> Tensor:
...
+        for name, param in m.named_parameters(recurse=False):
+            if param is None:
                 continue
 
             with torch.no_grad():
-                new_prm = Parameter(empty_like(prm), prm.requires_grad)
+                new_param = Parameter(empty_like(param), param.requires_grad)
 
-            setattr(m, name, new_prm)
+            setattr(m, name, new_param)
```
Comment (shagunsodhani) [inline src/fairseq2/nn/utils/module.py:140]: Curious what is the benefit of using `setattr(m, name, new_param)` over `m.name = new_param`

Comment (cbalioglu) [reply src/fairseq2/nn/utils/module.py:140]: `name` here is of type string. So syntactically you cannot write a valid Python statement like `m."foo" = "xyz"`.

Comment (shagunsodhani) [reply src/fairseq2/nn/utils/module.py:140]: got it - thank you :)

Comment (cbalioglu) [reply src/fairseq2/nn/utils/module.py:140]: no problem! :)

**Pull request merged**